### PR TITLE
New (enhanced?) redirection system

### PR DESCRIPTION
**Pull request by Emrio**
# New (enhanced?) redirection system

**This PR introduces breaking changes**

I propose a new redirection system which seems more straightforward for me at least.
I have also added the possibility to add a cooldown to the redirection.
Also, I felt that having a whole file to instantiate and execute the redirection is a bit overkill. I have therefore made the redirection in the HTML file itself.

This new system also takes care of both `fr` and `fr-FR` languages.
For instance, if we set up a redirection with these languages (order matters, first is the default): `en-US`, `fr-FR`, `fr-CA` and the user has accepted these languages: `jp-JP`, `fr`; he will be redirected to `fr-FR` because `fr-FR` is the first code in french.

### CODE REVIEW

File: src/js/redirection.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  }
+
+  /**
+   * @private @method _redirect
+   * @param  {String} lang       Language folder
+   * @param  {String} [url]      Optional sub URL
+   * @param  {Number} [cooldown] Optional timer cooldown before redirection (in seconds)
+   */
+  _redirect (lang, url, cooldown) {
+    setTimeout(function () {
```
Comment (Sukaato) [inline src/js/redirection.js]: `setTimeout()` is a arrow function like `foreach()`

File: src/js/redirection.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+   * @method go
+   * @param  {String} [url]      Optional sub URL - real URL will be 'folderLanguage/url'
+   * @param  {Number} [cooldown] Optional timer cooldown before redirection (in seconds)
+   * @example
+   * redirection.go('index.html') // Redirecting to /lang/index.html
+   * or
+   * redirection.go('index.html', 2) // Redirecting to /lang/index.html after 2 seconds
+   */
+  go (url, cooldown) {
+    const userlangs = window.navigator.languages || window.navigator.language || window.navigator.userLanguage
```
Comment (Sukaato) [inline src/js/redirection.js]: `navigator.languages` is available , so `window` prefix is useless

File: src/js/redirection.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+  /**
+   * @private @method _redirect
+   * @param  {String} lang       Language folder
+   * @param  {String} [url]      Optional sub URL
+   * @param  {Number} [cooldown] Optional timer cooldown before redirection (in seconds)
+   */
+  _redirect (lang, url, cooldown) {
+    setTimeout(function () {
+      window.location.href = `${lang}/${url || ''}`
```
Comment (Sukaato) [inline src/js/redirection.js]: same, `window` prefix is useless

File: src/js/redirection.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+   * @param  {String} [url]      Optional sub URL - real URL will be 'folderLanguage/url'
+   * @param  {Number} [cooldown] Optional timer cooldown before redirection (in seconds)
+   * @example
+   * redirection.go('index.html') // Redirecting to /lang/index.html
+   * or
+   * redirection.go('index.html', 2) // Redirecting to /lang/index.html after 2 seconds
+   */
+  go (url, cooldown) {
+    const userlangs = window.navigator.languages || window.navigator.language || window.navigator.userLanguage
+    for (const lang of Array.isArray(userlangs) ? userlangs : [ userlangs ]) {
```
Comment (Sukaato) [inline src/js/redirection.js:44]: Consider using consistent styling? `[ userlangs ]` for a array, and line 44 `[lang]` for a array too ?

**Review by Sukaato — CHANGES_REQUESTED** [review]

File: src/js/redirection.js (hunk 4 above)
Comment (Emrio) [reply src/js/redirection.js:44]: `[lang]` is not an array; it is a property accessor...

**Comment by Emrio** [comment]
@Sukaato I have updated `redirection.js` with your requested changes. Could you resolve the conflict in `index.html`?

**Pull request merged**